app/services/provider_authorisation.rb

```diff
@@ -103,39 +97,72 @@ class NotAuthorisedError < StandardError; end
 
 private
 
-  def ratifying_provider_can_view_safeguarding_information?(course:)
-    @actor.providers.include?(course.accredited_provider) &&
-      ProviderRelationshipPermissions.exists?(
-        ratifying_provider_can_view_safeguarding_information: true,
-        ratifying_provider: course.accredited_provider,
-        training_provider: course.provider,
-      )
+  def relationship_for(course:)
+    ProviderRelationshipPermissions.find_by(
+      training_provider: course.provider,
+      ratifying_provider: course.accredited_provider,
+    )
   end
 
-  def training_provider_can_view_safeguarding_information?(course:)
+  def permission_as_training_provider_user?(permission:, course:)
+    permission_name = "training_provider_can_#{permission}"
+    relationship = relationship_for course: course
+
     @actor.providers.include?(course.provider) &&
-      ProviderRelationshipPermissions.exists?(
-        training_provider_can_view_safeguarding_information: true,
-        ratifying_provider: course.accredited_provider,
-        training_provider: course.provider,
-      )
+      (relationship.blank? || relationship.send(permission_name))
   end
 
-  def ratifying_provider_can_view_diversity_information?(course:)
+  def permission_as_ratifying_provider_user?(permission:, course:)
+    permission_name = "ratifying_provider_can_#{permission}"
+    relationship = relationship_for course: course
+
     @actor.providers.include?(course.accredited_provider) &&
-      ProviderRelationshipPermissions.exists?(
-        ratifying_provider_can_view_diversity_information: true,
-        ratifying_provider: course.accredited_provider,
-        training_provider: course.provider,
-      )
+      (relationship.blank? || relationship.send(permission_name))
   end
 
-  def training_provider_can_view_diversity_information?(course:)
-    @actor.providers.include?(course.provider) &&
-      ProviderRelationshipPermissions.exists?(
-        training_provider_can_view_diversity_information: true,
-        ratifying_provider: course.accredited_provider,
-        training_provider: course.provider,
-      )
+  def full_authorisation?(permission:, course:)
+    return true if @actor.is_a?(SupportUser)
+
+    training_provider = course.provider
+    ratifying_provider = course.accredited_provider
+
+    # enforce user-level permissions
+    errors << :requires_provider_user_permission unless
+      @actor.is_a?(VendorApiUser) ||
```

Comment:
Should we be checking the `ApiUser`'s permissions here too?

Reply:
`VendorApiUser`s do not have user-level permissions, all that matters for them is their provider association and checking the `course_option_id` they supply for `make_decisions` belongs to them. Their provider association, of course, may bring in org-level restrictions.